src/api-v1.js

```diff
@@ -99,7 +99,15 @@ api.declare({
       // instead of a more complicated structure
 
       if (x === 0 && (result.status === 'absent' || result.status === 'error')) {
-        let validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
+        let validUrl;
+        try {
+          validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
```

Comment:
validate first checks that the URL matches the allowed regexes, then does all the redirects as HEAD requests, validates that all are HTTPS instead of HTTP and then returns.  It only throws when there is an error trying to do a HEAD request.  That's the codepath we're changing here: if we through during a request, we send back the error's status code to set the request's status code.

The case where the URL is invalid, but we don't have any http errors is handled by a falsy return value, and handling for that is unchanged by this PR